src/emulator/modules/SceLibKernel/SceLibKernel.cpp

```diff
@@ -88,30 +88,31 @@ EXPORT(int, sceClibMemchr) {
     return UNIMPLEMENTED();
 }
 
-EXPORT(int, sceClibMemcmp) {
-    return UNIMPLEMENTED();
+EXPORT(int, sceClibMemcmp, const void *ptr1, const void *ptr2, SceSize num) {
+    return memcmp(ptr1, ptr2, num);
 }
 
 EXPORT(int, sceClibMemcmpConstTime) {
     return UNIMPLEMENTED();
 }
 
-EXPORT(int, sceClibMemcpy, void *destination, const void *source, SceSize num) {
-    memcpy(destination, source, num);
-    return 0;
+EXPORT(Ptr<void>, sceClibMemcpy, Ptr<void> destination, const void *source, SceSize num) {
```

Comment:
why is source not a `Ptr<>`? you sure it works?

Reply:
Original code wasn't a Ptr too.

Reply:
Those functions are all wrong. Not sure why they originally work. Use ptr<>, it represents guest's 32 bit pointer. Using those is basically 64-bit integer.

Reply:
There are a lot of functions using direct C pointers in the whole emulator codebase. I can recall @petmac a long time ago saying it being now safe after a PR due to changes in bridging.

Reply:
Sorry I didn't remember. These are avail since forever. [Here](https://github.com/Vita3K/Vita3K/blob/master/src/emulator/module/include/module/bridge_types.h#L34)

Reply:
Unless we need the guest address, we can safely use this